pkg/handler/common/provider/gcp.go

```diff
@@ -0,0 +1,329 @@
+	clusterProvider := ctx.Value(middleware.ClusterProviderContextKey).(provider.ClusterProvider)
+	cluster, err := handlercommon.GetCluster(ctx, projectProvider, privilegedProjectProvider, userInfoGetter, projectID, clusterID, &provider.ClusterGetOptions{CheckInitStatus: true})
+	if err != nil {
+		return nil, err
+	}
+	if cluster.Spec.Cloud.GCP == nil {
+		return nil, errors.NewNotFound("cloud spec for ", clusterID)
+	}
+
+	assertedClusterProvider, ok := clusterProvider.(*kubernetesprovider.ClusterProvider)
+	if !ok {
+		return nil, errors.New(http.StatusInternalServerError, "failed to assert clusterProvider")
+	}
+
+	secretKeySelector := provider.SecretKeySelectorValueFuncFactory(ctx, assertedClusterProvider.GetSeedClusterAdminRuntimeClient())
+	sa, err := gcp.GetCredentialsForCluster(cluster.Spec.Cloud, secretKeySelector)
+	if err != nil {
+		return nil, err
+	}
+	userInfo, err := userInfoGetter(ctx, "")
+	if err != nil {
+		return nil, common.KubernetesErrorToHTTPError(err)
+	}
+	return ListGCPSubnetworks(ctx, userInfo, cluster.Spec.Cloud.DatacenterName, sa, network, seedsGetter)
+}
+
+func GCPDiskTypesWithClusterCredentialsEndpoint(ctx context.Context, userInfoGetter provider.UserInfoGetter, projectProvider provider.ProjectProvider, privilegedProjectProvider provider.PrivilegedProjectProvider, projectID, clusterID, zone string) (interface{}, error) {
+	clusterProvider := ctx.Value(middleware.ClusterProviderContextKey).(provider.ClusterProvider)
+
+	cluster, err := handlercommon.GetCluster(ctx, projectProvider, privilegedProjectProvider, userInfoGetter, projectID, clusterID, &provider.ClusterGetOptions{CheckInitStatus: true})
+	if err != nil {
+		return nil, err
+	}
+	if cluster.Spec.Cloud.GCP == nil {
+		return nil, errors.NewNotFound("cloud spec for ", clusterID)
+	}
+
+	assertedClusterProvider, ok := clusterProvider.(*kubernetesprovider.ClusterProvider)
+	if !ok {
+		return nil, errors.New(http.StatusInternalServerError, "failed to assert clusterProvider")
+	}
+
+	secretKeySelector := provider.SecretKeySelectorValueFuncFactory(ctx, assertedClusterProvider.GetSeedClusterAdminRuntimeClient())
+	sa, err := gcp.GetCredentialsForCluster(cluster.Spec.Cloud, secretKeySelector)
+	if err != nil {
+		return nil, err
+	}
+
+	return ListGCPDiskTypes(ctx, sa, zone)
+}
+
+func ListGCPDiskTypes(ctx context.Context, sa string, zone string) (apiv1.GCPDiskTypeList, error) {
+	diskTypes := apiv1.GCPDiskTypeList{}
+
+	computeService, project, err := gcp.ConnectToComputeService(sa)
+	if err != nil {
+		return diskTypes, err
+	}
+
+	req := computeService.DiskTypes.List(project, zone)
+	err = req.Pages(ctx, func(page *compute.DiskTypeList) error {
+		for _, diskType := range page.Items {
+			if diskType.Name != "local-ssd" {
+				// TODO: There are some issues at the moment with local-ssd, that's why it is disabled at the moment.
```

Comment:
is this still actual? maybe we should add a ticket to check thi.

Reply:
I will create an issue for this. Let leave this like it is right now